prime-router/src/main/kotlin/common/LoggerExt.kt

```diff
@@ -0,0 +1,45 @@
+package gov.cdc.prime.router.common
+
+import org.apache.logging.log4j.ThreadContext
+import org.apache.logging.log4j.kotlin.KotlinLogger
+
+/**
+ * Collection of logger extension functions that accept a context map along with the message
+ *
+ * The context map fields will show up in the "customDimensions" collection in azure in the "traces"
+ * or "exceptions" table (depending on if an exception was passed)
+ */
+fun KotlinLogger.trace(msg: Any, context: Map<String, String>) = withContext(context) {
```

Comment:
Can we add follow up ticket, but I think it would be great to either add an overload that would allow passing in values like `Map<String, SomeSerializableType>`?  Mostly thinking it would be great to do something like `logger.info("Starting FHIR Convert", mapOf("message", FhirConvertMessage))`

Reply:
Something like that is in the works for my events PR.

Unfortunately it appears that Azure (and `ThreadContext`) only accept a flat String to String map. The workaround there is to serialize a class to JSON and use it as the value. The upside there is with a few extra lines of query in the log explorer it can be extracted out into a new queryable column. The downside is the query is less efficient and requires parsing.

I'm building the PRs off each other so if this is something we want before merging it might make sense to combine the PRs.